### PR TITLE
[READY] new mutation: autotomy

### PR DESCRIPTION
**Pull request by Lett1**
## About The Pull Request

Adds a new mutation to genetics: Autotomy. This allows a creature to shed a limb on command, mostly when in danger to distract predators.

## Why It's Good For The Game

More genetics is allows fun. Use it to escape cuffs but watch out, losing a leg when you want an arm gone is not good. Combine with ligament hook for wacky shenanigans.

Watch it in action [here](https://streamable.com/x6qdy)

## Changelog
:cl:
add: Genetics has discovered a new mutation: Autotomy!
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
Would be neat if you could target it.

**Comment by tralezab** [comment]
Usually things with autotomy... have ways to get the limbs back...

**Comment by Lett1** [comment]
> Usually things with autotomy... have ways to get the limbs back...

Surgery and ligment hook my man

**Comment by tralezab** [comment]
true!

**Comment by Lett1** [comment]
> Would be neat if you could target it.

It's random to make using it to escape cuffs more risky

**Comment by Tlaltecuhtli** [comment]
make the limb animate

**Comment by PKPenguin321** [comment]
Nice, I was planning to add this but also with the ability to pop limbs back in. [It was gonna be a combination power](https://file.house/62Et.png). If I get around to finishing the rest of the powers on the list, maybe I'll do that to this if that's cool

File: code/datums/mutations/actions.dm [hunk 1]
```diff
@@ -190,3 +190,38 @@
...
+	text_gain_indication = "<span class='notice'>Your joints feel loose.</span>"
+	instability = 30
+	power = /obj/effect/proc_holder/spell/self/self_amputation
+
+/obj/effect/proc_holder/spell/self/self_amputation
+	name = "Drop a limb"
+	desc = "Concentrate to make a random limb pop right off your body."
+	clothes_req = FALSE
+	human_req = FALSE
+	charge_max = 100
```
Comment (PKPenguin321) [inline code/datums/mutations/actions.dm]: Should have a really really small recharge time so you can just shoot limbs off in every direction at will

**Comment by knacker48** [comment]
You could make so if you combine it with the synchronizer chromosome you can choose the limb. The chromosome system is already underutilized as it is so it couldn't hurt to make it a bit more relevant

**Comment by Lett1** [comment]
> You could make so if you combine it with the synchronizer chromosome you can choose the limb. The chromosome system is already underutilized as it is so it couldn't hurt to make it a bit more relevant

Actually a good idea, gonna look into how these work.

**Comment by CRITAWAKETS** [comment]
You should make the power chromosone shoot off everyone of your limbs but at incredible high speed while doing high damage to anybody in its path.

**Comment by Ratvar** [comment]
Now replace changeling's acid uncuff with this

**Comment by Lett1** [comment]
Well, seems like chromosome fuckery doesn't really extent to the action itself so thats off the table for now.

**Comment by Time-Green** [comment]
Chromosomes are like stat modifiers. They aren't an on/off switch. Still, you could make it do the selection whenever the stabilizer stat is above a certain threshold

File: code/datums/mutations/actions.dm [hunk 2]
```diff
@@ -190,3 +190,38 @@
...
+			if(BP.body_part != HEAD && BP.body_part != CHEST)
+				if(BP.dismemberable)
+					parts += BP
+		if(!parts.len)
+			to_chat(usr, "<span class='notice'>You can't shed any more limbs!</span>")
+			return
+		var/obj/item/bodypart/BP = pick(parts)
+		BP.dismember()
+		return
+	. = ..()
```
Comment (nemvar) [inline code/datums/mutations/actions.dm]: On the topic of early returns, I don't think parent does anything here.

File: code/datums/mutations/actions.dm [hunk 3]
```diff
@@ -190,3 +190,38 @@
...
+/obj/effect/proc_holder/spell/self/self_amputation
+	name = "Drop a limb"
+	desc = "Concentrate to make a random limb pop right off your body."
+	clothes_req = FALSE
+	human_req = FALSE
+	charge_max = 100
+	action_icon_state = "autotomy"
+
+/obj/effect/proc_holder/spell/self/self_amputation/cast(mob/user = usr)
+	if(iscarbon(user))
```
Comment (nemvar) [inline code/datums/mutations/actions.dm]: Easy early return

File: code/datums/mutations/actions.dm [hunk 4]
```diff
@@ -190,3 +190,38 @@
 /obj/effect/proc_holder/spell/self/void/cast(mob/user = usr)
 	. = ..()
 	new /obj/effect/immortality_talisman/void(get_turf(user), user)
+
+/datum/mutation/human/self_amputation
+	name = "Autotomy"
+	desc = "Allows a creature to voluntary discard a random appendage."
+	quality = MINOR_NEGATIVE //upsides and downsides
```
Comment (nemvar) [inline code/datums/mutations/actions.dm]: What exactly is the downside? There doesn't appear to be one unless you press the button.

**Comment by optimumtact** [comment]
power creep

**Comment by PKPenguin321** [comment]
Hardly powercreep. I think I've explained this before, but with instability and the way new genetics works, you can't just have every power if you can even get them all (which you almost never will)

**Comment by tralezab** [comment]
or you COULD do that, IF you had it require gene mixing with works but is woefully underused at the moment

**Comment by ATH1909** [comment]
> make the limb animate

This; I want to send endless armies of limbs against my enemies as a slimeperson (or changeling)!

**Pull request merged**